### PR TITLE
Add databricks environment to pyspark-iris starter

### PR DESCRIPTION
**Pull request by AhdraMeraliQB**
## Motivation and Context
 [Kedro #2722](https://github.com/kedro-org/kedro/issues/2272) highlights a bug introduced in #82. The consensus to remedy this was to introduce a databricks environment with specified dbfs filepaths as paths were not being resolved correctly.

## How has this been tested?
Following [the instructions in the docs](https://kedro.readthedocs.io/en/stable/deployment/databricks.html), with the addition of specifying the environment used. Relevant changes have been made to the docs in this PR. 

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Assigned myself to the PR
- [ ] Added tests to cover my changes

### CODE REVIEW

File: pyspark-iris/{{ cookiecutter.repo_name }}/conf/databricks/catalog.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    header: True
+    inferSchema: True
+  save_args:
+    header: True
+
+# We need to set mode to 'overwrite' in save_args so when saving the dataset it is replaced each time it is run
+# for all SparkDataSets.
+X_train@pyspark:
+  type: spark.SparkDataSet
+  filepath: data/02_intermediate/X_train.parquet
```
Comment (jmholzer) [inline pyspark-iris/{{ cookiecutter.repo_name }}/conf/databricks/catalog.yml:59]:
```suggestion
  filepath: /dbfs/root/projects/iris-databricks/data/02_intermediate/X_train.parquet
```

File: pyspark-iris/{{ cookiecutter.repo_name }}/conf/databricks/catalog.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+# The Data Catalog supports being able to reference the same file using two different DataSet implementations
+# (transcoding), templating and a way to reuse arguments that are frequently repeated. See more here:
+# https://kedro.readthedocs.io/en/stable/data/data_catalog.html
+#
+# This is a data set used by the iris classification example pipeline provided with this starter
+# template. Please feel free to remove it once you remove the example pipeline.
+
+example_iris_data:
+  type: spark.SparkDataSet
+  filepath: data/01_raw/iris.csv
```
Comment (jmholzer) [inline pyspark-iris/{{ cookiecutter.repo_name }}/conf/databricks/catalog.yml:47]:
```suggestion
  filepath: /dbfs/root/projects/iris-databricks/data/01_raw/iris.csv
```

Our recommendation in our reworked docs is to always use the `/dbfs` prefix when working with DBFS, even though it isn't strictly necessary when saving. See the discussion on [this ticket](https://github.com/kedro-org/kedro-plugins/pull/114) if you're interested in the rationale 🙂.

File: pyspark-iris/{{ cookiecutter.repo_name }}/conf/databricks/catalog.yml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  save_args:
+    mode: overwrite
+
+X_train@pandas:
+  type: pandas.ParquetDataSet
+  filepath: /dbfs/root/projects/iris-databricks/data/02_intermediate/X_train.parquet
+
+X_test@pyspark:
+  type: spark.SparkDataSet
+  filepath: data/02_intermediate/X_test.parquet
```
Comment (jmholzer) [inline pyspark-iris/{{ cookiecutter.repo_name }}/conf/databricks/catalog.yml:69]:
```suggestion
  filepath: /dbfs/root/projects/iris-databricks/data/02_intermediate/X_test.parquet
```

File: pyspark-iris/{{ cookiecutter.repo_name }}/conf/databricks/catalog.yml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  save_args:
+    mode: overwrite
+
+X_test@pandas:
+  type: pandas.ParquetDataSet
+  filepath: /dbfs/root/projects/iris-databricks/data/02_intermediate/X_test.parquet
+
+y_train@pyspark:
+  type: spark.SparkDataSet
+  filepath: data/02_intermediate/y_train.parquet
```
Comment (jmholzer) [inline pyspark-iris/{{ cookiecutter.repo_name }}/conf/databricks/catalog.yml:79]:
```suggestion
  filepath: /dbfs/root/projects/iris-databricks/data/02_intermediate/y_train.parquet
```

File: pyspark-iris/{{ cookiecutter.repo_name }}/conf/databricks/catalog.yml [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  save_args:
+    mode: overwrite
+
+y_train@pandas:
+  type: pandas.ParquetDataSet
+  filepath: /dbfs/root/projects/iris-databricks/data/02_intermediate/y_train.parquet
+
+y_test@pyspark:
+  type: spark.SparkDataSet
+  filepath: data/02_intermediate/y_test.parquet
```
Comment (jmholzer) [inline pyspark-iris/{{ cookiecutter.repo_name }}/conf/databricks/catalog.yml:89]:
```suggestion
  filepath: /dbfs/root/projects/iris-databricks/data/02_intermediate/y_test.parquet
```

**Review by jmholzer — COMMENTED** [review]
Awesome work! 😃 Thanks for this, I really like this environment implementation.

I just left one suggestion, then we're good to go.

**Review by noklam — COMMENTED** [review]
Thank you for this PR, it saves me sometime to setup the repository on Databricks.

I found a couple of issues but not all are related to this PR, so I will discuss them separately.
I think you may need to update `logging.yml` as well as the default logger create problems for me, have you tried running it on Databricks?

```
OSError: [Errno 30] Read-only file system: '/Workspace/Repos/nok_lam_chan@xxxx.com/pyspark-on-databricks.ide/logs/info.log'
```

There are still problem though @AntonyMilneQB 
1. You need `%load_ext kedro.ipython` which loads `%reload_kedro` by default
2. `%reload_kedro --env=databricks` - hoping to override the log
3. You get error, because 1. setup filebase logging already, any log created by plugin/framework will hit the filesystem before you can actually reset the log

```python
import logging
logging_config = {"version":1}
logging.config.dictConfig(logging_config)
```

I need to run this before 2. to load up the correct environment.

**Comment by AntonyMilneQB** [comment]
@noklam this is a great spot 💯 

I've made [db20207](https://github.com/kedro-org/kedro-starters/pull/119/commits/db202076ac88baf03995a6f0640f54a6f0a932ef) which should fix the issue - please can you confirm?

**Comment by noklam** [comment]
@AntonyMilneQB I can confirm it's working now, thanks!

File: pyspark-iris/{{ cookiecutter.repo_name }}/conf/base/logging.yml [hunk 6]
```diff
@@ -44,4 +44,8 @@ loggers:
     level: INFO
 
 root:
-  handlers: [rich, info_file_handler, error_file_handler]
+  # File-based logging is disabled by default so that the starter works immediately on Databricks Repos.
+  # See https://docs.kedro.org/en/stable/logging/logging.html#disable-file-based-logging for more.
+  # To enable file-based logging, change the below line to:
+  # handlers: [rich, info_file_handler, error_file_handler]
+  handlers: [rich]
```
[the remark this answers is not in the document]
Comment (AntonyMilneQB) [reply pyspark-iris/{{ cookiecutter.repo_name }}/conf/base/logging.yml:51]: Not sure I understand the question?

The line `handlers: [rich]` should still be there if that's what you mean 🙂  It's basically replacing `handlers: [rich, info_file_handler, error_file_handler]`.

[the remark this answers is not in the document]
Comment (AhdraMeraliQB) [reply pyspark-iris/{{ cookiecutter.repo_name }}/conf/base/logging.yml:51]: That's exactly what I meant! Thanks for clarifying, just wanted to be sure 😄

**Review by jmholzer — APPROVED** [review]
Excellent work, thanks Ahdra!

**Review by noklam — APPROVED** [review]
LGTM!

**Pull request merged**